templates/todo/projects/csharp-sql/README.md

```diff
@@ -33,123 +21,192 @@ The following prerequisites are required to use this application.  Please ensure
-```
+#### Run and Debug Locally
 
-### Clean up
+The easiest way to run and debug is to leverage the Azure Developer CLI Visual Studio Code Extension. Refer to this [walk-through](https://aka.ms/azure-dev/vscode) for more details.
 
-#### Deprovision Azure Resources
+#### Clean up resources
 
-Run the following command to delete the application from your Azure Subscription.
+When you are done, you can delete all the Azure resources created with this template by running the following command:
 
 ```bash
-az dev down
+azd down
 ```
 
-#### Remove the Azure Dev CLI extension
+### Additional azd commands
 
-Run the following command to remove the Azure Dev CLI extension:
+The Azure Developer CLI includes many other commands to help with your Azure development experience. You can view these commands at the terminal by running `azd help`. You can also view the full list of commands on our [Azure Developer CLI command](https://aka.ms/azure-dev/ref) page.
+
+## Troubleshooting/Known issues
+
+Sometimes, things go awry. If you happen to run into issues, then please review our ["Known Issues"](https://aka.ms/azure-dev/knownissues) page for help. If you continue to have issues, then please file an issue in our main [Azure Dev](https://aka.ms/azure-dev/issues) repository.
+
+## Security
+
+### Roles
+
+This template creates a [managed identity](https://docs.microsoft.com/azure/active-directory/managed-identities-azure-resources/overview) for your app inside your Azure Active Directory tenant, and it is used to authenticate your app with Azure and other services that support Azure AD authentication like Key Vault via access policies. You will see principalId referenced in the infrastructure as code files, that refers to the id of the currently logged in Azure CLI user, which will be granted access policies and permissions to run the application locally. To view your managed identity in the Azure Portal, follow these [steps](https://docs.microsoft.com/azure/active-directory/managed-identities-azure-resources/how-to-view-managed-identity-service-principal-portal).
+
+The template will also create a dedicated Azure SQL user for the application and store its credentials in a KeyVault (see below). The application user's access is restricted to the ToDo database only; a separate administrator account is set up for managing the whole Azure SQL instance. 
+
+### Key Vault
+
+This template uses [Azure Key Vault](https://docs.microsoft.com/azure/key-vault/general/overview) to securely store your Azure SQL Database credentials and the connection string used by the application to interact with the database. Key Vault is a cloud service for securely storing and accessing secrets (API keys, passwords, certificates, cryptographic keys) and makes it simple to give other Azure services access to them. As you continue developing your solution, you may add as many secrets to your Key Vault as you require.
+
+## Uninstall
+
+To uninstall the Azure Developer CLI:
```

Comment:
Again, not sure that this really belongs here. Thinking more about azd install instructions, this also won't scale once we support package managers for install.

Reply:
Will be handled via https://github.com/Azure/azure-dev/issues/599